WordPress/Classes/Services/MediaServiceRemoteCoreREST.swift

```diff
@@ -0,0 +1,217 @@
+import Foundation
+import Combine
+import WordPressCore
+import WordPressShared
+import WordPressAPI
+import WordPressAPIInternal
+import WordPressKit
+
+/// A `MediaServiceRemote` implementation that uses the WordPress core REST API (`/wp-json/wp/v2/media`).
+class MediaServiceRemoteCoreREST: NSObject, MediaServiceRemote {
+    let client: WordPressClient
+
+    init(client: WordPressClient) {
+        self.client = client
+    }
+
+    func getMediaWithID(_ mediaID: NSNumber, success: ((RemoteMedia?) -> Void)?, failure: (((any Error)?) -> Void)?) {
+        Task { @MainActor in
+            do {
+                let media = try await client.api.media.retrieveWithEditContext(mediaId: mediaID.int64Value).data
+                success?(RemoteMedia(media: media))
+            } catch {
+                failure?(error)
+            }
+        }
+    }
+
+    func uploadMedia(
+        _ media: RemoteMedia,
+        progress progressPtr: AutoreleasingUnsafeMutablePointer<Progress?>?,
+        success: ((RemoteMedia?) -> Void)?,
+        failure: (((any Error)?) -> Void)?
+    ) {
+        wpAssert(media.localURL != nil)
```

Comment:
(nit) Add `guard let localURL = media.localURL` to avoid the need for a force cast.

Reply:
I have updated them to throw an arbitrary error (which should never happen) when nil. https://github.com/wordpress-mobile/WordPress-iOS/pull/24540/commits/c225088eb886304a3efd6b72fd8d72837132c0cf